compiler/rustc_const_eval/src/const_eval/machine.rs

```diff
@@ -745,6 +749,12 @@ impl<'mir, 'tcx> interpret::Machine<'mir, 'tcx> for CompileTimeInterpreter<'mir,
         // Everything else is fine.
         Ok(())
     }
+
+    fn before_alloc_read(ecx: &InterpCx<'mir, 'tcx, Self>, alloc_id: AllocId) {
+        if Some(alloc_id) == ecx.machine.static_root_alloc_id {
+            ecx.tcx.dcx().emit_err(crate::errors::RecursiveStatic { span: ecx.tcx.span });
```

Comment:
This continues evaluation after displaying an error. Why is that a good idea? It'll see uninit data and then likely things will go wrong.

Seems better to raise an `InterpError`? We have `ConstEvalErrKind` for errors that can only happen in the const-eval machine.